ding/policy/r2d2.py

```diff
@@ -404,62 +513,104 @@ def _forward_collect(self, data: dict, eps: float) -> dict:
         return {i: d for i, d in zip(data_id, output)}
 
     def _reset_collect(self, data_id: Optional[List[int]] = None) -> None:
+        """
+        Overview:
+            Reset some stateful variables for eval mode when necessary, such as the hidden state of RNN or the \
+            memory bank of some special algortihms. If ``data_id`` is None, it means to reset all the stateful \
+            varaibles. Otherwise, it will reset the stateful variables according to the ``data_id``. For example, \
+            different environments/episodes in evaluation in ``data_id`` will have different hidden state in RNN.
+        Arguments:
+            - data_id (:obj:`Optional[List[int]]`): The id of the data, which is used to reset the stateful variables \
+                (i.e., RNN hidden_state in R2D2) specified by ``data_id``.
+        """
         self._collect_model.reset(data_id=data_id)
 
-    def _process_transition(self, obs: Any, model_output: dict, timestep: namedtuple) -> dict:
-        r"""
+    def _process_transition(self, obs: torch.Tensor, policy_output: Dict[str, torch.Tensor],
+                            timestep: namedtuple) -> Dict[str, torch.Tensor]:
+        """
         Overview:
-            Generate dict type transition data from inputs.
+            Process and pack one timestep transition data info a dict, which can be directly used for training and \
+            saved in replay buffer. For R2D2, it contains obs, action, prev_state, reward, done.
         Arguments:
-            - obs (:obj:`Any`): Env observation
-            - model_output (:obj:`dict`): Output of collect model, including at least ['action', 'prev_state']
-            - timestep (:obj:`namedtuple`): Output after env step, including at least ['reward', 'done'] \
-                (here 'obs' indicates obs after env step).
+            - obs (:obj:`torch.Tensor`): The env observation of current timestep, such as stacked 2D image in Atari.
+            - policy_output (:obj:`Dict[str, torch.Tensor]`): The output of the policy network with the observation \
+                as input. For R2D2, it contains the action and the prev_state of RNN.
+            - timestep (:obj:`namedtuple`): The execution result namedtuple returned by the environment step method, \
+                except all the elements have been transformed into tensor data. Usually, it contains the next obs, \
+                reward, done, info, etc.
```

Comment:
- obs: The environment observation at the current timestep (e.g., a stacked 2D image in Atari).
- policy_output: The output from the policy network given the observation as input. For R2D2, this includes the action and the ``prev_state`` of the RNN.
- timestep: The result returned by the environment's step method, with all elements transformed 
            into tensor data. Usually, this includes the ``next_obs``, ``reward``, ``done``, ``info``(other information).